lib/common/http/request.ts

```diff
@@ -287,6 +287,11 @@ export const makeSingleRawRequestToDownstream = async (
         logger.error({ error }, 'Error making request to downstream.');
         reject(error);
       });
+      request.on('timeout', () => {
```

Comment:
Out of curiosity, what was causing this loop to fail to timeout when we were trying to debug this with the debugger?

Reply:
it was something about `setTimeout` that we were using that didn't play it right